packages/builders/src/messages/Message.ts

```diff
@@ -335,7 +337,7 @@ export class MessageBuilder implements JSONEncodable<RESTPostAPIChannelMessageJS
 	public setComponents(
 		...components: RestOrArray<
 			| ActionRowBuilder
-			| APIActionRowComponent<APIMessageActionRowComponent>
+			| APIActionRowComponent<APIComponentInMessageActionRow>
```

Comment:
Either needs to accept all top-level component types (ugly) or simply be removed. Neither `ActionRowBuilder` nor `ChatInputCommandBuilder` have a `set` method for "replacing" the existing components/options. That can already be achieved with `spliceComponents()` after all.

Reply:
I could see us supporting a setComponents with all types but I feel like that's just horrible, and should probably stick to add (maybe add in a clear method so you can start with a blank slate)?

But I'd ask @didinele's thoughts too on this. I could easily implement a setComponents i think

Reply:
Sorry, I'm a bit lost on what the problem is here

Reply:
The problem is that this is a generic "set the components array to this arbitrary list of components" when we have specific `addXComponent` methods for all the other ways. As already stated we don't have a set method in ActionRowBuilder or ChatInputCommandBuilder or ContainerBuilder or SectionBuilder. So why would we suddenly have one in MessageBuilder?

Reply:
I see. I added it with the thought that there's only ActionRows so it sort of made sense. I'm for removing it in this case.